src/ariel-os-coap/src/transport_slipmux.rs

```diff
@@ -0,0 +1,337 @@
+            Option<(
+                embassy_sync::pipe::Reader<
+                    'static,
+                    embassy_sync::blocking_mutex::raw::CriticalSectionRawMutex,
+                    { MAX_FULL_SPEED_PACKET_SIZE as usize },
+                >,
+                cdc_acm::Sender<'static, ariel_os_embassy::usb::UsbDriver>,
+            )>,
+        >,
+    > = once_cell::sync::OnceCell::new();
+
+    pub(super) async fn take() -> impl embedded_io_async::Read + embedded_io_async::Write {}
+}
+
+mod over_uart {
+    use super::*;
+
+    use ariel_os_embassy::hal::peripherals;
+    use ariel_os_hal::{define_peripherals, uart};
+
+    type UartAssignment = ariel_os_boards::pins::HOST_FACING_UART;
+
+    // FIXME There has to be a better idiom for a function deep down to take peripherals that does
+    // not involve spawning a task that terminates at startup?
+
+    static TAKE_THIS: OnceLock<core::cell::RefCell<Option<UartAssignment>>> = OnceLock::new();
+
+    #[ariel_os_macros::task(autostart, peripherals)]
+    async fn pass_on_uart(peripherals: UartAssignment) {
+        TAKE_THIS
+            .init(Some(peripherals).into())
+            .ok()
+            .expect("Autostart is only called once");
+    }
+
+    // FIXME: Move to BufRead once <https://github.com/ariel-os/ariel-os/pull/1600> is resolved.
+    pub(super) async fn take() -> impl embedded_io_async::Read + embedded_io_async::Write {
+        let peripherals: UartAssignment = TAKE_THIS.try_get().unwrap().take().unwrap();
+
+        static RX_BUF: ConstStaticCell<[u8; 16]> = ConstStaticCell::new([0; 16]);
+        static TX_BUF: ConstStaticCell<[u8; 16]> = ConstStaticCell::new([0; 16]);
+        let rx_buf = RX_BUF.take();
+        let tx_buf = TX_BUF.take();
+
+        let mut config = ariel_os_hal::hal::uart::Config::default();
+        config.baudrate = ariel_os_hal::uart::Baudrate::_115200;
+
+        use ariel_os_hal::uart::Assignment;
+        let (tx, rx) = peripherals.into_pins();
+
+        let mut uart =
+            <<MainUartPins as Assignment>::Device<'_>>::new(rx, tx, rx_buf, tx_buf, config)
+                .expect("Invalid UART configuration");
+
+        uart
+    }
+}
+
+pub(crate) async fn coap_run_slipmux(mut handler: impl coap_handler::Handler) -> ! {
+    use embedded_io_async::{Read, Write};
+    use slipmux::{DecodeStatus, Decoder};
+
+    // For the time being we play the simple game, where we just have a server, and can afford to
+    // read until something has arrived, then write out what has been written, and continue.
+    //
+    // This will not be sustainable with the next generation coap-handler, and then we may need to
+    // split the UARTs.
+
+    //let uart = usb_serial::take().await;
+    let mut uart = over_uart::take().await;
+
+    let mut decoder = slipmux::Decoder::new();
+    // Small -- just enough so we can show some bytes for diagnostics. (We can do larger if
+    // ReferencedLatestFrame learns to use a single buffer).
```

Comment:
You can also always implement your own handler. No need to use one of the pre-builts if they don't suit your usecase. :) 

```Rust
pub trait FrameHandler {
    // Required methods
    fn begin_frame(&mut self, frame_type: FrameType);
    fn write_byte(&mut self, byte: u8);
    fn end_frame(&mut self, error: Option<Error>);
}
```

Reply:
Trying that out in fb754dad8604c502687d71ac038d95d14c7c1303. Turns out all I need to store is a heapless vec and a byte for overflow, because while I don't get the info on what kind of frame ended in `end_frame`, I don't need that there, and I get it from where I feed the byte in too.